TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import 'Styles/Camera/Calibrate.scss';
+import styles from 'Styles/Camera/Calibrate.scss';
+
+import React, { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { ConnectionManager } from 'TouchFree/Connection/ConnectionManager';
+import { InputActionManager } from 'TouchFree/Plugins/InputActionManager';
+import { InputType, InteractionType, TouchFreeInputAction } from 'TouchFree/TouchFreeToolingTypes';
+
+import {
+    CalibrationHandLostMessage,
+    CalibrationCancelButton,
+    CalibrationInstructions,
+    CalibrationTutorialVideo,
+    CalibrationProgressCircle,
+} from './CalibrationComponents';
+
+const { handNotFoundHeight } = styles;
+
+interface CalibrationScreenProps {
+    isHandPresent: boolean;
+    onCancel: () => void;
+}
+
+export const TIMEOUT_S = 10;
+const TIMEOUT_MS = TIMEOUT_S * 1000;
+const INITIAL_WAIT_MS = 5000;
+
+const CalibrationScreen: React.FC<CalibrationScreenProps> = ({ isHandPresent, onCancel }) => {
+    const [displayHandIndicator, setDisplayHandIndicator] = React.useState(false);
+    // ===== Click Progress =====
+    const [progressToClick, setProgressToClick] = React.useState(0);
+    const [progress, setProgress] = React.useState(0);
+    const isNewClick = React.useRef(false);
+    // ===== Timeout =====
+    const [timeToPosSelect, setTimeToPosSelect] = React.useState(TIMEOUT_S);
+    const timeout = React.useRef<NodeJS.Timeout>();
+    const interval = React.useRef<NodeJS.Timer>();
+    const [initialWaitOver, setInitialWaitOver] = React.useState(false);
+    // ===== React Router =====
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const handleClick = (path: string) => {
+        ConnectionManager.serviceConnection()?.QuickSetupRequest(
+            path === '../bottom' ? true : false,
+            () => {},
+            () => {}
+        );
+        navigate(path);
+    };
+
+    const handleTFInput = (evt: CustomEvent<TouchFreeInputAction>): void => {
+        if (evt.detail.InteractionType === InteractionType.HOVER) {
+            if (!isNewClick.current) {
+                isNewClick.current = evt.detail.ProgressToClick === 0;
+                return;
+            }
+
+            if (evt.detail.InputType === InputType.MOVE || evt.detail.InputType === InputType.DOWN) {
+                setProgressToClick(evt.detail.ProgressToClick);
+            }
+        }
+    };
+
+    useEffect(() => {
```

Comment:
Could these two useEffect be combined into one as they are both on start-up (no dependencies) or do you prefer them split? (personally prefer them grouped by dependencies so in the same useEffect)

Reply:
I think when it's this small it makes sense to combine. But I do like having them separate if there's a few large chunks of code that need the return statements.